src/use_hotkeys.rs

```diff
@@ -1,140 +1,130 @@
-            }
+#[cfg(any(feature = "hydrate", feature = "csr"))]
+use wasm_bindgen::JsValue;
 
-            if hotkey.modifiers.shift {
-                modifiers_match &= pressed_keyset.contains_key("shift");
-            }
+#[cfg(any(feature = "hydrate", feature = "csr"))]
+use web_sys::KeyboardEvent;
 
-            if hotkey.modifiers.meta {
-                modifiers_match &= pressed_keyset.contains_key("meta");
-            }
+fn is_hotkey_match(hotkey: &Hotkey, pressed_keyset: &mut HashMap<String, KeyboardEvent>) -> bool {
+    let mut modifiers_match = true;
 
-            if hotkey.modifiers.alt {
-                modifiers_match &= pressed_keyset.contains_key("alt");
-            }
+    if hotkey.modifiers.ctrl {
+        modifiers_match &= pressed_keyset.contains_key("control");
+    }
 
-            if modifiers_match {
-                let keys_match = hotkey.keys.iter().all(|key| {
-                    if let Some(event) = pressed_keyset.get_mut(key) {
-                        event.prevent_default();
-                        true
-                    } else {
-                        false
-                    }
-                });
+    if hotkey.modifiers.shift {
+        modifiers_match &= pressed_keyset.contains_key("shift");
+    }
 
-                modifiers_match && keys_match
+    if hotkey.modifiers.meta {
+        modifiers_match &= pressed_keyset.contains_key("meta");
+    }
+
+    if hotkey.modifiers.alt {
+        modifiers_match &= pressed_keyset.contains_key("alt");
+    }
+
+    if modifiers_match {
+        let keys_match = hotkey.keys.iter().all(|key| {
+            if let Some(event) = pressed_keyset.get_mut(key) {
+                event.prevent_default();
+                true
             } else {
                 false
             }
-        }
+        });
 
-        pub fn use_hotkeys_scoped(
-            key_combination: String,
-            on_triggered: Callback<()>,
-            scopes: Vec<String>,
-        ) {
-            let parsed_keys: HashSet<Hotkey> = key_combination
-                .split(',')
-                .map(|key_combo| Hotkey::new(key_combo))
-                .collect();
+        modifiers_match && keys_match
+    } else {
+        false
+    }
+}
 
-            let hotkeys_context = use_hotkeys_context();
-            let pressed_keys = hotkeys_context.pressed_keys;
+pub fn use_hotkeys_scoped(
+    key_combination: String,
+    on_triggered: Callback<()>,
+    scopes: Vec<String>,
+) {
+    let parsed_keys: HashSet<Hotkey> = key_combination
+        .split(',')
+        .map(|key_combo| Hotkey::new(key_combo))
+        .collect();
+
+    let hotkeys_context = use_hotkeys_context();
+    let pressed_keys = hotkeys_context.pressed_keys;
+
+    create_effect(move |_| {
+        let active_scopes = hotkeys_context.active_scopes.get();
+        let within_scope = &scopes.iter().any(|scope| active_scopes.contains(scope));
+
+        if *within_scope {
```

Comment:
This looks weird but maybe there is a reason for that. It shouldn't be...?

```suggestion
        let within_scope = scopes.iter().any(|scope| active_scopes.contains(scope));

        if within_scope {
```